howto/port/install-as-system-app.md

```diff
@@ -0,0 +1,70 @@
+Anbox Cloud allows user applications to be installed as Android system apps. This can be done by using the `aam install-system-app` command.
+
+### Make a system app
+To be a system app running in the system process of Android system, you must add the attribute `android:sharedUserId="android.uid.system"` into the `<manifest>` or `<application>` tag in the AndroidManifest.xml file of your Android project.
+
+```xml
+...
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    package="<package_name>"
+    android:sharedUserId="android.uid.system">
+...
+```
+
+Then build and sign the application alongside with other Android applications when building your Android image or sign it with [Android Studio](https://developer.android.com/studio/publish/app-signing). For the latter approach, the system platform key is not required and you can use the keys that are generated from Android Studio to sign the application.
+
+### Create an addon
+
+By creating an addon and invoking the `aam install-system-app` command in a `pre-start` hook, the signed APK can be installed as a system app in the Android container.
+
+Following the [tutorial](https://discourse.ubuntu.com/t/creating-an-addon/25284) and create the basic layout of an addon, which has a manifest.yaml file and hooks folder prepared in advance. Assumed that the APK file is named `app.apk`, place the APK into the addon folder and create a `pre-start` hook with the following content.
```

Comment:
Not sure I understand "which has a manifest.yaml file and hooks folder prepared in advance". Do you mean the user should prepare this as described in the tutorial?

Reply:
Yes. that's what a basic skeleton of an addon looks like.
We were thinking in the past to introduce a command like 
- `amc addon create`
- `amc application create` 
to provide an interactive way for end user on an application/addon creation. Just like `lxd init` does today
```
$ sudo lxd init
Name of the storage backend to use (dir or zfs) [default=zfs]: zfs
Create a new ZFS pool (yes/no) [default=yes]? no
Name of the existing ZFS pool or dataset: lxd
Would you like LXD to be available over the network (yes/no) [default=no]? no
Do you want to configure the LXD bridge (yes/no) [default=yes]? no
```
So after a user answered all questions, we have a manifest.yaml file well prepared with all requirements filled up to create an app/addon. In this way, the app/addon creation would be more intuitive than what it works today.
(I guess we still have it in our backlog.)

Reply:
I think we should show the full addon here rather than referencing another howto. That would make this self contained, ideas?